atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/bucketingthings/DefaultBucketCloseTimestampCalculator.java

```diff
@@ -91,9 +91,16 @@ public OptionalLong getBucketCloseTimestamp(long startTimestamp) {
         }
 
         long finalLogicalTimestamp = puncherStore.get(closeWallClockTimeMillis);
+        long clampedTimestamp = clampTimestampToCoarsePartitionBoundary(finalLogicalTimestamp);
 
-        // if this case happens, it's possibly clockdrift or a delayed write.
-        if (finalLogicalTimestamp <= startTimestamp) {
+        // if this case happens, it's possibly clockdrift or a delayed write, or we just make too slow progress
+        // TODO(mdaudali): Re-evaluate if the lower bound of 10_000_000 timestamps per 10 mins makes sense.
+        // This has the following interesting case: Suppose that the assigner stopped for a day. Then, suppose that the
+        // within the following 10 minute period from where the assigner currently is looking at, there is _not_ 10 mil
+        // timestamps. Then, instead of choosing a 10 million sized block, we have a giant block up to the fresh Ts.
+        // This is capped at 5 billion, and we were okay with this case when there's clock drift, but we should
+        // evaluate.
```

Comment:
I think it's fine given that it's capped at 5B. In theory it could be a problem for services that write massive transactions only when they run, and only run very infrequently (though that's a very very specific case).

Reply:
That should also be addressed by the config in the next PR - for smaller services, we can have their cap be much lower.